common/workunit/workunit.cpp

```diff
@@ -13611,6 +13611,18 @@ extern WORKUNIT_API void addTimeStamp(IWorkUnit * wu, StatisticScopeType scopeTy
     wu->setStatistic(queryStatisticsComponentType(), queryStatisticsComponentName(), scopeType, scopestr, kind, NULL, getTimeStampNowValue(), 1, 0, StatsMergeAppend);
 }
 
+extern WORKUNIT_API double calculateThorCost(__int64 timeNs, unsigned clusterWidth)
+{
+    IPropertyTree *costs = queryCostsConfiguration();
+    if (costs)
+    {
+        double thor_master_rate = costs->getPropReal("cost[@name='master']/@value", 0.0);
```

Comment:
One comment from before:

It is worth thinking what it will look like in the yaml file to configure it.  Currently you have:

```
- cost:
      name: master
      value:  10.2 
- cost:
      name: slave
      value:  12.3
```
With xpaths thor/@master and thor/@slave you would have
```
costs:
  thor:
    master: 10.2
    slave: 12.3
```
which is cleaner.  (I think we are going to change our elements to not have a leading upper case letter for compatibility with normal k8s conventions now that we are reading yaml files directly).

Reply:
Now that Jake's thor changes are merged, it would be worth adding code in the yaml files to provide values into the thor.yaml configuration file.

Reply:
similar recommendations here: https://helm.sh/docs/topics/chart_best_practices/values/#consider-how-users-will-use-your-values